src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -14,89 +14,207 @@
 // limitations under the License.
 //*****************************************************************************
 #include <algorithm>
-#include <cmath>
 #include <cstddef>
 #include <cstring>
 #include <iostream>
 #include <limits>
 #include <string>
 #include <vector>
+#include <cmath>
+#include <numeric>
+
+#include "opencv2/opencv.hpp"
 
 #include "../../custom_node_interface.h"
-#include "opencv2/core.hpp"
-#include "opencv2/highgui.hpp"
-#include "opencv2/imgcodecs.hpp"
-#include "opencv2/imgproc.hpp"
+#include "nms.hpp"
+#include "utils.hpp"
 
 static constexpr const char* IMAGE_TENSOR_NAME = "image";
 static constexpr const char* SCORES_TENSOR_NAME = "scores";
 static constexpr const char* GEOMETRY_TENSOR_NAME = "geometry";
 static constexpr const char* TEXT_IMAGES_TENSOR_NAME = "text_images";
+static constexpr const char* COORDINATES_TENSOR_NAME = "text_coordinates";
+static constexpr const char* CONFIDENCE_TENSOR_NAME = "confidence_levels";
 
 #define NODE_ASSERT(cond, msg)                       \
     if (!(cond)) {                                   \
         std::cout << "Assert: " << msg << std::endl; \
         return 1;                                    \
     }
 
-cv::Mat nchw_to_mat(const CustomNodeTensor* input) {
-    // std::vector<float> data(input->dataLength / sizeof(float));
-    // std::cout << "Before: " << data.size() << std::endl;
-    // uint64_t C = input->dims[1];
-    // uint64_t H = input->dims[2];
-    // uint64_t W = input->dims[3];
-    // std::cout << "C: " << C << "; H: " << H << "; W: " << W << std::endl;
-    // for(uint64_t c = 0; c < C; ++c) {
-    //     for(uint64_t h = 0; h < H; ++h) {
-    //         for(uint64_t w = 0; w < W; ++w) {
-    //             uint64_t nchwOffset = (c * H * W) + (h * W) + w;
-    //             uint64_t nhwcOffset = (h * W * C) + (w * C) + c;
-    //             data[nhwcOffset] = input->data[nchwOffset];
-    //         }
-    //     }
-    // }
-    uint64_t H = input->dims[1];
-    uint64_t W = input->dims[2];
-    cv::Mat image(H, W, CV_32FC3);
-    // std::memcpy((void*)image.data, (void*)data.data(), input->dataLength);
-    std::memcpy((void*)image.data, (void*)input->data, input->dataLength);
-    return image;
+template<typename T>
```

Comment:
It would be good to move common functions to utils.hpp. They could be more reusable that way.

Reply:
Done